bundles/org.openhab.binding.mqtt.homie/src/main/java/org/openhab/binding/mqtt/homie/internal/homie300/Property.java

```diff
@@ -194,7 +194,7 @@ public void createChannelFromAttribute() {
                 if (step != null && !isDecimal && step.intValue() <= 0) {
                     step = new BigDecimal(1);
                 }
-                if (attributes.unit != null && attributes.unit.contains("%")) {
+                if (attributes.unit.contains("%") && attributes.settable) {
```

Comment:
You got rid of the `attributes.unit` null check. Please add it back in.

Reply:
Hi @cpmeister! The null check is not needed, because it is initialized as an [empty string](https://github.com/openhab/openhab-addons/blob/2.5.x/bundles/org.openhab.binding.mqtt.homie/src/main/java/org/openhab/binding/mqtt/homie/internal/homie300/PropertyAttributes.java#L54) :wink: